rust/abacus-base/src/settings/chains.rs

```diff
@@ -222,16 +239,34 @@ impl ChainSetup<InboxAddresses> {
     }
 
     /// Get a clone of the metrics conf with correctly configured contract information.
-    pub fn metrics_conf(&self) -> PrometheusMiddlewareConf {
+    pub fn metrics_conf(
+        &self,
+        agent_name: &str,
+        signer: &Option<Signers>,
+    ) -> PrometheusMiddlewareConf {
         let mut cfg = self.metrics_conf.clone();
+
+        if cfg.chain.is_none() {
```

Comment:
Is there an easy way to dedup?

Reply:
I don't know if I understand what you are asking, but basically if they override the config, then it will not get set here.